pkg/kubectl/apply/strategy/replace_visitor.go

```diff
@@ -98,3 +109,42 @@ func (v replaceStrategy) doAddOrDelete(e apply.Element) (apply.Result, bool) {
 }
 
 var _ apply.Strategy = &replaceStrategy{}
+
+func (v replaceStrategy) detectConflict(e apply.Element) (apply.Result, error) {
+	switch e.(type) {
+	case apply.ListElement:
```

Comment:
The behavior for each of these should be documented.  What does it mean to detect a conflict for a list?  It shouldn't be just if the elements have conflicts, it should also check if there are any elements missing or added.